src/chapters/angular-lsn2/ngFor.rst

```diff
@@ -0,0 +1,207 @@
+``ngFor``
+==========
+
+In the :ref:`Angular lesson 1 exercises <angular-exercises-1>`, you modified
+a ``movie-list`` component to display a series of titles. The final code
+within ``movie-list.component.html`` probably looked something like:
+
+.. sourcecode:: html
+   :linenos:
+
+   <div class='movies'>
+      <h3>Movies to Watch</h3>
+      <ol>
+         <li>{{movies[0]}}</li>
+         <li>{{movies[1]}}</li>
+         <li>{{movies[2]}}</li>
+         <li>{{movies[3]}}</li>
+      </ol>
+   </div>
+
+``movies[0]`` - ``movies[3]`` reference an array assigned within the
+``movie-list.component.ts`` file.
+
+To change the number of movie titles displayed in the ordered list, we could
+manually add or remove ``li`` tags, or we could use the structural directive
+``ngFor`` to iterate through the movie options.
+
+``ngFor`` Syntax
+-----------------
+
+The example below shows how to use ``ngFor`` to iterate through the contents of
+an array. It is the basic approach, and ``ngFor`` can be used in more complex
+ways. For a complete guide to using ``ngFor`` and all of its variations, refer
+to the following resources:
+
+#. `Angular documentation <https://angular.io/guide/template-syntax#ngFor>`__,
+#. `Malcoded website <https://malcoded.com/posts/angular-ngfor/>`__.
+
+Just like a ``for`` loop in JavaScript requires a specific syntax in order to
+operate, loops in Angular must follow a set of rules. Let's explore these rules
+by adding ``ngFor`` to our movie list code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <div class='movies'>
+      <h3>Movies to Watch</h3>
+      <ol>
+         <li *ngFor = "let movie of movies">{{movie}}</li>
+      </ol>
+   </div>
+
+Some items to note:
+
+#. Structural directives all begin with the ``*`` symbol.
+#. The string ``"let movie of movies"`` provides the instructions
+   for running the loop.
+
+   a. The ``let`` keyword declares the ``movie`` variable.
+   b. ``of movies`` sets ``movie`` equal to the first element of the ``movies``
+      array. Each iteration of the loop sets ``movie`` equal to the next title
+      in the array.
+
+#. The ``*ngFor`` statement is placed INSIDE the ``<li>`` tag.
+#. ``{{movie}}`` is the placeholder for the current value of ``movie``.
+
+By placing the ``*ngFor`` statement inside the tag, the loop generates
+multiple ``<li></li>`` pairs. Each iteration adds a new list item to the
```

Comment:
how do you feel about `multiple <li> elements.`? I'm slightly concerned about using the term "pairs".

Reply:
Fixed. Thanks.